File: generated_tasks/pantry__3c79d183b353/tests/hidden/base_repo/src/format/color.ts
// Tiny ANSI color helpers. Honors NO_COLOR and TERM=dumb. No third-party deps.

const RESET = "\x1b[0m";
const CODES = {
  bold: "\x1b[1m",
  dim: "\x1b[2m",
  red: "\x1b[31m",
  green: "\x1b[32m",
  yellow: "\x1b[33m",
  blue: "\x1b[34m",
  cyan: "\x1b[36m",
  magenta: "\x1b[35m",
} as const;

export type ColorName = keyof typeof CODES;

let disabled = computeDisabled();

function computeDisabled(): boolean {
  if (process.env.NO_COLOR) return true;
  if (!process.env.TERM || process.env.TERM === "dumb") return true;
  return false;
}

/** Force colors off; for tests / pipes. */
export function disable(): void {
  disabled = true;
}

/** Force colors on; for tests. */
export function enable(): void {
  disabled = false;
}

export function isEnabled(): boolean {
  return !disabled;
}

function wrap(code: string, s: string): string {
  if (disabled) return s;
  return `${code}${s}${RESET}`;
}

export function bold(s: string): string {
  return wrap(CODES.bold, s);
}

export function dim(s: string): string {
  return wrap(CODES.dim, s);
}

export function red(s: string): string {
  return wrap(CODES.red, s);
}

export function green(s: string): string {
  return wrap(CODES.green, s);
}

export function yellow(s: string): string {
  return wrap(CODES.yellow, s);
}

export function blue(s: string): string {
  return wrap(CODES.blue, s);
}

export function cyan(s: string): string {
  return wrap(CODES.cyan, s);
}

export function magenta(s: string): string {
  return wrap(CODES.magenta, s);
}

export function colorize(name: ColorName, s: string): string {
  return wrap(CODES[name], s);
}

File: generated_tasks/kindling__args-positional/tests/hidden/base_repo/internal/color/color.go
// Package color provides ANSI color helpers used by interactive
// kindling output.
package color

import "os"

// Code is an ANSI SGR sequence.
type Code string

const (
	Reset   Code = "\x1b[0m"
	Bold    Code = "\x1b[1m"
	Dim     Code = "\x1b[2m"
	Red     Code = "\x1b[31m"
	Green   Code = "\x1b[32m"
	Yellow  Code = "\x1b[33m"
	Blue    Code = "\x1b[34m"
	Magenta Code = "\x1b[35m"
	Cyan    Code = "\x1b[36m"
	Gray    Code = "\x1b[90m"
)

// Enabled reports whether color output should be used. Honors NO_COLOR.
func Enabled() bool {
	return os.Getenv("NO_COLOR") == ""
}

// Wrap wraps s in c + reset, or returns s unchanged if disabled.
func Wrap(c Code, s string) string {
	if !Enabled() {
		return s
	}
	return string(c) + s + string(Reset)
}

// LevelColor returns the color associated with a log level token.
func LevelColor(level string) Code {
	switch level {
	case "error", "fatal":
		return Red
	case "warn", "warning":
		return Yellow
	case "info":
		return Green
	case "debug":
		return Cyan
	default:
		return Gray
	}
}

// Disable turns color output off for the remainder of the process.
// Used in tests.
func Disable() {
	_ = os.Setenv("NO_COLOR", "1")
}

// Enable turns color output back on.
func Enable() {
	_ = os.Unsetenv("NO_COLOR")
}
